### PR TITLE
Create codeql-analysis.yml

### PR DESCRIPTION
**Pull request by dopey**

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'cpp', 'python' ]
```
Comment (maraino) [inline .github/workflows/codeql-analysis.yml]: Is `cpp` used for both C and C++, because the code is in regular C. 

And I don't see any python file.

**Review by maraino — CHANGES_REQUESTED** [review]
There's no python.

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (dopey) [reply .github/workflows/codeql-analysis.yml]: https://codeql.github.com/docs/codeql-overview/supported-languages-and-frameworks/ -- according the codeql docs they do support regular C. But I don't think that's a language you can select. I think `cpp` selects C as well (according to the extensions).

**Review by maraino — DISMISSED** [review]

**Comment by maraino** [comment]
@dopey The build looks like is failing, probably because missing dependencies, we actually build this in landlord. See https://github.com/smallstep/landlord/blob/69ff14a21de9e184cd47f9627d491f5907ed71db/deploy/docker/Dockerfile#L1-L10

**Review by maraino — APPROVED** [review]

**Pull request merged**